oak_functions/loader/src/metrics.rs

```diff
@@ -120,19 +118,23 @@ impl OakApiNativeExtension for PrivateMetricsExtension<Logger> {
 /// Provides logic for the host ABI function [`report_metric`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#report_metric).
 fn report_metric(
     extension: &mut PrivateMetricsExtension<Logger>,
-    raw_label: Vec<u8>,
-    value: i64,
+    request: Vec<u8>,
 ) -> Result<(), OakStatus> {
-    let label = std::str::from_utf8(raw_label.as_slice()).map_err(|err| {
+    let request: ReportMetricRequest =
+        bincode::deserialize(&request).expect("Fail to deserialize report metric request.");
+
+    let label = std::str::from_utf8(request.raw_label.as_slice()).map_err(|err| {
```

Comment:
As mentioned this should no longer be required with the serialisation logic.

Reply:
Done.